### ISSUE TITLE
Allow to r+ branch from a commit message

### ISSUE THREAD

**Issue by matklad** [bors-ng/bors-ng#243]
Often one needs to push a really small one-line change, like a typo fix. Currently, the possible workflows are

1) Push to a branch, create pull request, open GitHub UI, add a self-r+ comment. 

2) Push directly to master.

1) is a bit to heavy weight for trivial changes, and 2) violates the not rocket science rule (even a typo fix may break the build) and cancels all in-flight bors pull request.

It would be cool if it were possible to add some special `bors instantr+` message to the commit message itself, push it to a branch, and have it merged by bors automatically without creating a PR.

**Comment by ishitatsuyuki** [comment]
We could have an "instant" branch utilizing git's separate configuration for push and pull, so it's easy to sync to master while the push is staged.

The failure can be either delivered as direct email or GitHub-comment based notification.

**Comment by notriddle** [comment]
The best way I can think to do it would be to do it exactly how the original bors did it (even though homu doesn't do it this way):

Commit comments (try scrolling to the bottom of a commit to see the UI of it) should be able to have bors commands run on them just like real Pull Requests can. The only difference is that you don't burn an issue number (and, since a commit is identified by its hash, you can't change a commit like you can change a PR).

**Comment by matklad** [comment]
>Commit comments

The comments in GitHub UI? This still seems to require opening GitHub UI though... But it is indeed more lightweight then the PR workflow.

Why not allow r+ in commit message? One of the possible drawbacks is that if you rebase, cherry-pick or otherwise change the hash of the commit, confusion will arise. But presumably these should not happen in the intended use-case of one-line changes?

**Comment by ishitatsuyuki** [comment]
My proposal even removes the need to r+, just making a "master-like" branch but verified. @notriddle how do you think about approach? Is explicit approval preferred, even if it's just a self approval?

**Comment by notriddle** [comment]
Right now, bors-ng doesn't listen to the `push` event. Having to add that is really the big disadvantage of both approaches (especially since there are a *lot* of push events on your typical busy repo).

Commit comments, on the other hand, are rare, so turning it on would be pretty cheap.

I'm probably falling into status quo bias, though. We already listen for the issue_comment event, which is also a very busy event.

**Comment by ishitatsuyuki** [comment]
I believe there's more issue comments than pushes due to the review pattern enforced by bors. Both events require a repository config query, so they are almost identically expensive. Anyway, are we really going to overload the server with human events? We may need to consider other optimizations first.

**Comment by notriddle** [comment]
Yeah, I probably am being overly cautious. The `push` event wouldn't introduce any problems that we don't already deal with. We just need to get sharding to work on the public instance.

---

> I believe there's more issue comments than pushes due to the review pattern enforced by bors.

Half of the comments bors receives are on Issues, not Pull Requests. Bors always ignores non-PR Issue comments, but GitHub provides no way to prevent them from coming in. They would be unaffected by any workflow changes.

**Comment by notriddle** [comment]
Also, I would probably prefer explicit self-approval. Explicit is better than implicit, all else being equal.

**Comment by HiddenKn** [comment]
I don't like the idea of bors commands through comments because I get a lot of commits from upstream who might also start using bors at some point but I don't want them to just cause my bors to approve all their changes!
